tomcat-7-valves/src/main/java/org/jboss/openshift/valves/StdoutAccessLogValve.java

```diff
@@ -16,31 +16,1111 @@
 
 package org.jboss.openshift.valves;
 
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.InetAddress;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpSession;
+
+import org.apache.catalina.AccessLog;
+import org.apache.catalina.Globals;
+import org.apache.catalina.LifecycleException;
+import org.apache.catalina.LifecycleState;
+import org.apache.catalina.Session;
+import org.apache.catalina.connector.Request;
+import org.apache.catalina.connector.Response;
 import org.apache.catalina.valves.AccessLogValve;
+import org.apache.catalina.valves.Constants;
+import org.apache.catalina.valves.ValveBase;
+import org.apache.coyote.RequestInfo;
+import org.apache.juli.logging.Log;
+import org.apache.juli.logging.LogFactory;
+import org.apache.tomcat.util.ExceptionUtils;
 
 /**
  * @author <a href="mailto:rmartine@redhat.com">Ricardo Martinelli</a>
  */
-public class StdoutAccessLogValve extends AccessLogValve {
-
-	@Override
-	public boolean isRotatable() {
-		return false;
-	}
-
-	@Override
-	public void rotate() {
-		// Do nothing
-	}
-
-	@Override
-	public synchronized boolean rotate(String newFileName) {
-		return true;
-	}
-
-	@Override
-	public void log(String message) {
-		System.out.println(message);
-	}
-	
+public class StdoutAccessLogValve extends ValveBase implements AccessLog {
+
+    private static final Log log = LogFactory.getLog(AccessLogValve.class);
+
+    public StdoutAccessLogValve() {
+        super(true);
+    }
+
+    private volatile String dateStamp = "";
+
+    protected static final String info =
+        "org.apache.catalina.valves.AccessLogValve/2.2";
+
+    protected boolean enabled = true;
+
+    protected String pattern = null;
+
+    private boolean buffered = true;
+
+    protected PrintWriter writer = null;
+
+    protected SimpleDateFormat fileDateFormatter = null;
+
+    private static final int globalCacheSize = 300;
+
+    private static final int localCacheSize = 60;
+
+    protected File currentLogFile = null;
```

Comment:
All references to files should be removed.  It might be better to start with ValveBase and implement the methods declared on AccessLog from scratch, using AccessLogValve as a guide (or AbstractAccessLogValve from tomcat8 as a base).